app_legacy/Helpers/database/player-rank.php

```diff
@@ -162,4 +153,6 @@ function getUserRank(string $username, int $type = RankType::Hardcore): ?int
 
         return (int) legacyDbFetch($query)['UserRank'];
     });
+
+    return $cachedRank ?? null;
```

Comment:
isn't `?? null` redundant?

Reply:
Yeah, that makes no sense anymore :D 

That was a leftover from when I tried to make sure that any cached string value is cast to int, too, for `declare(strict_types=1)` not causing `TypeError`s.

Reverted to what you initially had written to directly return the cached value.